src/publisher/mqtt_publisher.py

```diff
@@ -50,14 +50,19 @@ async def connect(self) -> None:
                 )
             else:
                 self.client.set_auth_credentials(username=self.configuration.mqtt_user)
+
         if self.transport_protocol.with_tls:
             cert_uri = self.configuration.tls_server_cert_path
             LOG.debug(
                 f"Configuring network encryption and authentication options for MQTT using {cert_uri}"
             )
             ssl_context = ssl.SSLContext()
-            ssl_context.load_verify_locations(cafile=cert_uri)
-            ssl_context.check_hostname = False
+            if cert_uri:
+                ssl_context.load_verify_locations(cafile=cert_uri)
+                ssl_context.check_hostname = False
```

Comment:
I wonder if this is a good idea. I'd rather see a new settings to override it. @bj00rn I'll merge as is but I'd like a new PR where we can configure this flag, defaulting it to True

We'll update the changelog accordingly noting the breaking change